### PR TITLE
Timeline fast scrolling & scroll zooming

### PR DESCRIPTION
**Pull request by Sublimity**

### CODE REVIEW

File: Quaver.Shared/Screens/Editor/EditorScreen.cs [hunk 1]
```diff
@@ -454,46 +454,45 @@ public void ChangeBeatSnap(Direction direction)
...
-        ///     arrow keys
+        ///     Handles seeking through the audio whether with the scroll wheel or arrow keys
         /// </summary>
         private void HandleAudioSeeking()
         {
-            if (AudioEngine.Track.IsStopped || AudioEngine.Track.IsDisposed || KeyboardManager.CurrentState.IsKeyDown(Keys.LeftShift)
-                || KeyboardManager.CurrentState.IsKeyDown(Keys.RightShift))
+            var view = (EditorScreenView) View;
+
+            if (AudioEngine.Track.IsStopped || AudioEngine.Track.IsDisposed || view.LayerCompositor.ScrollContainer.InputEnabled ||
```
Comment (Swan) [inline Quaver.Shared/Screens/Editor/EditorScreen.cs]: Put the duplicated code into a method.

File: Quaver.Shared/Screens/Editor/EditorScreen.cs [hunk 2]
```diff
@@ -478,7 +478,7 @@ private void HandleAudioSeeking()
                 return;
 
             var beatSnapValue = KeyboardManager.CurrentState.IsKeyDown(Keys.LeftControl) || KeyboardManager.CurrentState.IsKeyDown(Keys.RightControl)
```
Comment (Swan) [inline Quaver.Shared/Screens/Editor/EditorScreen.cs]: You should never nest ternary operators. It makes it more difficult to read.

File: Quaver.Shared/Screens/Editor/EditorScreen.cs [hunk 3]
```diff
@@ -469,7 +469,7 @@ private void HandleAudioSeeking()
         {
             var view = (EditorScreenView) View;
 
-            if (AudioEngine.Track.IsStopped || AudioEngine.Track.IsDisposed || view.LayerCompositor.ScrollContainer.InputEnabled ||
+            if (IsStoppedDisposedAndInputEnabled ||
```
Comment (Swan) [inline Quaver.Shared/Screens/Editor/EditorScreen.cs]: Move the duplicated keyboard state checks into its own method

**Comment by Sublimity** [comment]
Stale

**Pull request closed without merging**